### PR TITLE
WatchOS Architecture

### PR DESCRIPTION
**Pull request by JaviSoto**
Much like iOS and MacOSX, watchOS needs its own target...

### CODE REVIEW

**Comment by tonyd256** [comment]
@JaviSoto Thanks for putting this together! I tried to compile, but it looks like we're going to need Runes to support this before it will work.

**Comment by JaviSoto** [comment]
Yes! Sorry about that, should've marked this as WIP. I'm working on doing this for Runes as well :)



--

Javier Soto

**Comment by tonyd256** [comment]
Awesome thanks!

**Comment by tonyd256** [comment]
@JaviSoto I added a watchOS target to Runes but how did you get a test target on Argo-watchOS? There is not test target for watchOS that I could find.

**Comment by JaviSoto** [comment]
OK this is ready now :) Sorry for the delay, I've been busy since WWDC!
The work was already done in Runes: https://github.com/thoughtbot/Runes/commit/3a46bd92d78061c76c08a0b2baf6ee214dba77e9

File: Argo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -121,6 +133,15 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXCopyFilesBuildPhase section */
+		D03579C61B2B7A14005D26AE /* CopyFiles */ = {
+			isa = PBXCopyFilesBuildPhase;
+			buildActionMask = 2147483647;
+			dstPath = "";
+			dstSubfolderSpec = 10;
+			files = (
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: is this build phase doing anything? can it be removed?

File: Argo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -469,6 +508,25 @@
...
+				D03579C31B2B7A14005D26AE /* Headers */,
+				D03579C51B2B7A14005D26AE /* Resources */,
+				D03579C61B2B7A14005D26AE /* CopyFiles */,
+			);
+			buildRules = (
+			);
+			dependencies = (
+			);
+			name = "Argo-watchOS";
+			productName = "Argo-Mac";
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: Can this be the same as the name `Argo-watchOS`

File: Argo.xcodeproj/project.pbxproj (hunk 1 above)
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: actually nvm ... I'm assuming this was created by default when the watchOS target was created. If so, then just leave it.

File: Argo.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -771,6 +854,53 @@
...
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_VERSION = A;
+				GCC_PREPROCESSOR_DEFINITIONS = (
+					"DEBUG=1",
+					"$(inherited)",
+				);
+				INFOPLIST_FILE = "$(SRCROOT)/Argo/Resources/Info.plist";
+				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.9;
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: I don't think this is right. Was this target created from a copy of the Mac target?

I'm comparing the diff here with the one used for Runes fyi: https://github.com/thoughtbot/Runes/commit/6dc0d19d1049da768e3256ad3a39bbdef89f5e6d

File: Argo.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -771,6 +854,53 @@
...
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				DEFINES_MODULE = YES;
+				DYLIB_COMPATIBILITY_VERSION = 1;
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_VERSION = A;
+				INFOPLIST_FILE = "$(SRCROOT)/Argo/Resources/Info.plist";
+				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.9;
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: same

File: Argo.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -771,6 +854,53 @@
...
+		D03579C81B2B7A14005D26AE /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = YES;
+				COMBINE_HIDPI_IMAGES = YES;
+				DEFINES_MODULE = YES;
+				DYLIB_COMPATIBILITY_VERSION = 1;
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_VERSION = A;
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: I also don't see this in the Runes diff when adding a watchOS target ... but I see this:
`TARGETED_DEVICE_FAMILY = 4;` which I think is missing from this diff

**Comment by tonyd256** [comment]
@JaviSoto Looks like this target might have been created by copying the Mac target. Might be easier to remove this target and create a new one for the watchOS.

**Comment by JaviSoto** [comment]
>@JaviSoto Looks like this target might have been created by copying the Mac target. Might be easier to remove this target and create a new one for the watchOS.

That's correct, it was easier to avoid having to do everything manually. I'll fix it!

**Comment by JaviSoto** [comment]
I'll close this to open a new PR.

**Comment by JaviSoto** [comment]
Reopened in https://github.com/thoughtbot/Argo/pull/192

**Pull request closed without merging**